### DISCUSSION TITLE
Supprimer grumphp ?

### DISCUSSION THREAD

**Discussion by SebSept** [friends-of-presta/fop_console#203] [Ideas]
On peut se faire la discussion en fr.

J'avais pour idée de virer grumphp, vu que ça fonctionne plus vraiment chez moi et pas seulement.
Mais @Kaudaj n'a pas de problème.

Mon idée était remplacer grumphp par un ensemble de scripts composer (comme je l'ai déjà fais, ça fonctionne bien pour ce que j'ai pu en faire).

**Comment by ghost** [comment]
Je l'aime bien le vieux, mais disons que j'ai CSfixer à la sauvegarde et je lance phpstan donc j'avoue que je ne vois pas trop l'utilité de le garder.

**Comment by tom-combet** [comment]
Perso je pense que je ne pourrai plus me passer de pre-commit à présent, en tout cas dans mes projets.
Il y aura toujours un moment où j'oublierai de lancer tel ou tel outil. Là on a que CSFixer et Stan, mais sur mes projets j'ai ESLint, PHPUnit et header-stamp en plus, bientôt autoindex si je fais une PR avec un --dry-run...
Alors on pourrait laisser aux développeurs le soin d'assurer leur syntaxe avec l'IDE ou manuellement. Mais si c'est intégré au projet, ça permet deux choses selon moi : ça encourage à respecter les coding standards (moins d'efforts pour le dev, tout est déjà prêt) et tout le monde a les mêmes résultats ("t'as quelle extension de csfixer toi ? ah oui nan elle est outdated celle là, moi j'ai celle-ci" et cie).
Après ça n'exclut pas de combiner les deux : utiliser les extensions pour avoir des retours pendant qu'on code, corriger en temps réel ou quand on save, et puis valider en final avec le pre-commit en cas d'oubli, de bug de l'extension, de différence de version entre le projet et l'extension etc. Un peu comme une validation côté client (IDE) et une validation côté serveur (pre-commit + github workflows).

Après que ce soit Composer ou GrumPHP qui s'en occupe, en vrai tant que ça marche...
J'ai une préférence pour le vieux parce que je trouve que c'est assez simple d'utilisation au final, tout en yaml. Et puis même si on l'appelle le vieux, il est toujours maintenu régulièrement ! 😛
@SebSept Tu peux nous redonner le lien vers les scripts composer que tu avais déjà fais, pour qu'on puisse voir la différence stp ?

**Reply by SebSept** [reply]
Ah ben non, on va pas supprimer les outils php de qualité !
La question, c'est bien est-ce qu'on fait ça avec Grumphp ou autrement.

Ici, tu as un exemple : 
https://github.com/SebSept/ps_dev_base/blob/92d2384cd978d8a26fddeb6db3df1accf0fc7827/composer.json#L32
ici aussi : https://github.com/SebSept/composer-starter/blob/9a68b3cba4f08fbedcda4900daf31ee225572b94/composer.json#L25
Ou encore, si tu l'utilises ps_dev_base, il va créer des scripts composer et hook git precommit.

Au final, au lieu de centraliser dans un yaml, ça centralise dans composer.json.

J'ai pas d'avis tranché, tant que ça fonctionne ça me va aussi.

**Reply by tom-combet** [reply]
Ah oui j'avais oublié phive ! 
Bah en vrai ça a l'air simple aussi si on l'utilise. Sinon je tente de réparer grum vu que c'est déjà set up et si c'est galère on part sur phive et composer ? En vrai, c'est quasi pareil, juste deux façons de faire. Après j'aime bien la configuration built-in et complète de grum, pour pleins de packages différents.

**Reply by SebSept** [reply]
[Phive](https://phar.io/), c'est pas mal, ça permet d'être indépendant de composer. Mais c'est pas la question vraiment ici 😝 .
Mais de choisir entre des scripts composer ou grumphp.
Si tu n'as de motivation particulièrement forte pour changer tout, faut pas le faire !

Donc, oui, je crois que remettre Grumphp serait la meilleure option... en espérant que ça puisse fonctionner si on l'utilise en commitant directement en ligne de commande ou via l'ide.

On part la dessus @friends-of-presta/fop-console-contributors ? et tu t'y colles @Kaudaj ?

**Reply by nenes25** [reply]
Yes je suis pour grump php aussi :)

**Reply by tom-combet** [reply]
Ah oui ok ok je comprends mieux, j'avais pas très bien regardé. C'est surtout ta commande [PreCommitHook](https://github.com/SebSept/ps_dev_base/blob/92d2384cd978d8a26fddeb6db3df1accf0fc7827/src/Command/SebSept/PrecommitHook.php) qui fait tout enfaite.
Bref ok ça marche on fait comme ça, je m'occupe de Grum, et puis on avisera en fonction. 👍

**Reply by SebSept** [reply]
#150

**Reply by ghost** [reply]
Il est beau en plus 
![Capture d’écran de 2021-11-10 10-35-44](https://user-images.githubusercontent.com/5064590/141088310-8201f4e6-f0ee-4281-8b98-f244a9eb5148.png)

**Comment by SebSept** [comment]
@friends-of-presta/fop-console-contributors vous en pensez quoi ? 
Vous voyiez ce que c'est grumphp ?

**Comment by jf-viguier** [comment]
Je ne connais pas je vous fait confiance

**Reply by SebSept** [reply]
C'est bien aimable 😄 
Et tu vois de quoi on parle ?
Tu as pu le voir en fonctionnement sur ta machine ?
ça passe bien ?

**Comment by jf-viguier** [comment]
Non, ça ne fonctionne pas chez moi

**Reply by SebSept** [reply]
c'est noté. merci pour le retour 👍🏼

**Reply by SebSept** [reply]
au passage, tu aurais pu répondre en 'thread' @jf-viguier , comme sur slack quoi 😄

**Reply by tom-combet** [reply]
si même le patron s'y met 🙄 ... 🤭

**Reply by jf-viguier** [reply]
MDR !!

**Comment by SebSept** [comment]
Donc, en résumé, il y a que @nenes25 qui est pour maintenir grumphp , c'est bien ça ?
On le vire ?

votez 👍🏼 pour le garder et 👎🏼 pour le virer. 
ping @okom3pom @Kaudaj  @jf-viguier

**Reply by tom-combet** [reply]
Dans sa dernière version, je suis pro-grum. Mais effectivement en l'état (en php 7.2), on dirait qu'il est pas assez mature.

**Reply by SebSept** [reply]
Si on a une config qui fonctionne bien, on peut le garder, mais sinon faut le virer...
